Wikipedia/Code/YearInReviewCoordinator.swift

```diff
@@ -181,6 +185,77 @@ final class YearInReviewCoordinator: NSObject, Coordinator {
         let number = NSNumber(value: readCount)
         return formatter.string(from: number) ?? String(readCount)
     }
+    
+    func personalizedSlide2Title(day: String) -> String {
+        let format = WMFLocalizedString(
+            "year-in-review-personalized-day-title-format",
+            value: "You read most on %1$@.",
+            comment: "Year in review, personalized slide title for users that displays the weekday they read most. %1$@ is replaced with the weekday."
+        )
+        
+        return String.localizedStringWithFormat(format, getLocalizedDay(day: day))
+    }
+    
+    func personalizedSlide2Subtitle(day: String) -> String {
+        let format = WMFLocalizedString(
+            "year-in-review-personalized-day-subtitle-format",
+            value: "You read the most articles on %1$@. It's clear that %1$@ are your prime day for exploring new content. Thanks for making the most of your reading time!",
+            comment: "Year in review, personalized slide subtitle for users that displays the weekday they read most. %1$@ is replaced with the weekday."
+        )
+        return String.localizedStringWithFormat(format, getLocalizedDay(day: day))
+    }
+    
+    func getLocalizedDay(day: String) -> String {
+        let localizedDay: String
+        switch day {
+        case "Sunday":
```

Comment:
Your most read day (`WMFPageViewDay`)'s `day` value are these strings only if the user's device is in English. When my device language is in Spanish:

```
let dayFormatter = Date.FormatStyle().weekday(.wide)
let dayOfWeek = timestamp.formatted(dayFormatter) //value from my testing today is "jueves" 
```

Instead of that code in WMFPageViewsDataController, you should find a way to convert the timestamp into a day of the week that is the same regardless of device language (like a value 1 - 7).

Reply:
Would it be a problem that some people start weeks on Sundays and some start on Mondays?

Reply:
Yeah could be, good catch! Maybe we could check Calendar.current.firstWeekday at this point? Something like:

```
func getLocalizedDay(day: Int) -> String {
        let localizedDay: String
        
        switch (Calendar.current.firstWeekday, day) {
        case (1, 1), (2, 7): // Sunday
            localizedDay = WMFLocalizedString(
                "year-in-review-day-sunday",
                value: "Sundays",
                comment: "Localized name for Sunday in plural form."
            )
        case (1, 2), (2, 1): // Monday
            localizedDay = WMFLocalizedString(
                "year-in-review-day-monday",
                value: "Mondays",
                comment: "Localized name for Monday in plural form."
            )
       ...etc
```

So if the Calendar's weekday value for the timestamp is 1, and the Calendar's firstWeekday value is 2 (meaning they start counting the week on Mondays), I think you need to display "Monday".